examples/cpp/content_filter/SubscriberApp.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file SubscriberApp.cpp
+ *
+ */
+
+#include "SubscriberApp.hpp"
+
+#include <stdexcept>
+
+#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/rtps/common/Locator.h>
+#include <fastdds/rtps/transport/UDPv4TransportDescriptor.h>
+
+#include "HelloWorldTypeObjectSupport.hpp"
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace content_filter {
+SubscriberApp::SubscriberApp(
+        const CLIParser::subscriber_config& config,
+        const std::string& topic_name)
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new HelloWorldPubSubType())
+    , filter_topic_(nullptr)
+    , stop_(false)
+{
+    // Set intraprocess to OFF
+    LibrarySettings library_settings;
```

Comment:
If the the former suggestion is accepted, we would enable or disable this (multicast and datasharing) depending on the `--writer-side` argument